modules/compute_instance/outputs.tf

```diff
@@ -19,8 +19,17 @@ output "instances_self_links" {
   value       = google_compute_instance_from_template.compute_instance.*.self_link
 }
 
+output "names" {
+  description = "List of compute instance names"
+  value       = google_compute_instance_from_template.compute_instance.*.name
+}
+
+output "ip_list_by_instance" {
+  description = "List of IP addresses for each instance by name"
+  value       = { for instance in google_compute_instance_from_template.compute_instance : instance["name"] => [for nic in instance["network_interface"] : nic["network_ip"]] }
```

Comment:
Wouldn't this specific usecase  be better served by using a [data](https://www.terraform.io/docs/providers/google/d/datasource_compute_instance.html) block?

Reply:
Ignorance on my end. I will resolve and close this PR